spec/active_job/queue_adapters/cloudtasker_adapter_spec.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require_relative '../../shared/active_job/instantiation_context'
+
+RSpec.describe ActiveJob::QueueAdapters::CloudtaskerAdapter do
+  include_context 'of Cloudtasker ActiveJob instantiation'
+
+  subject(:adapter) { described_class.new }
+
+  let :example_worker_double do
+    instance_double(
+      "#{described_class.name}::Worker",
+      example_worker_args
+    ).tap { |double| allow(double).to receive(:schedule) }
+  end
+
+  before do
+    allow(described_class::Worker).to receive(:new)
+      .and_return example_worker_double
+  end
+
+  shared_examples 'of instantiating a Cloudtasker Worker from ActiveJob' do
+    it 'instantiates a new CloudtaskerAdapter Worker for the given job' do
+      expect(described_class::Worker).to receive(:new).with example_worker_args
+
+      adapter.enqueue(example_job)
+    end
+  end
+
+  describe '#enqueue' do
+    include_examples 'of instantiating a Cloudtasker Worker from ActiveJob'
```

Comment:
You don't need this shared example because you're already testing that your doubles are receiving the `schedule` method with the right arguments. The schedule method can only be received by the double if the `allow` stub on `new` has been triggered.

Reply:
I think for both `#enqueue` and `#enqueue_at` 2 examples are needed to assert 2 different things:
- That the worker was instantiated with the correct parameters (i.e. which currently is done by calling `#build_worker`)
- That the worker was scheduled to execute when expected (as soon as possible in the case of `#enqueue`, and at the given time in `#enqueue_at`)

The shared example is needed, because it validates the worker is instanced exactly the same, regardless of the method used to enqueue the job.

Reply:
I may think of an alternate way to get rid of the `allow` stub (which I feel is not the way I liked)... but it's not clear to me how to re-implement this test.